benchbuild/projects/benchbuild/xz.py

```diff
@@ -1,62 +1,63 @@
 from plumbum import local
 
-from benchbuild import project
-from benchbuild.utils import compiler, download, run, wrapping
-from benchbuild.utils.cmd import cp, make, tar
+import benchbuild as bb
+from benchbuild.source import HTTP
+from benchbuild.utils.cmd import make, tar
 
 
-@download.with_wget({'5.2.1': 'http://tukaani.org/xz/xz-5.2.1.tar.gz'})
-class XZ(project.Project):
+class XZ(bb.Project):
     """ XZ """
+    VERSION = '5.2.1'
     NAME = 'xz'
     DOMAIN = 'compression'
     GROUP = 'benchbuild'
-    VERSION = '5.2.1'
-    SRC_FILE = 'xz.tar.gz'
 
-    testfiles = [
-        "text.html", "chicken.jpg", "control", "input.source", "liberty.jpg"
+    SOURCE = [
+        HTTP(remote={'5.2.1': 'http://tukaani.org/xz/xz-5.2.1.tar.gz'},
+             local='xz.tar.gz'),
+        HTTP(remote={'1.0': 'http://lairosiel.de/dist/compression.tar.gz'},
+             local='compression.tar.gz')
     ]
 
     def compile(self):
-        self.download()
-
-        tar('xfz', self.src_file)
+        xz_source = bb.path(self.source_of('xz.tar.gz'))
+        xz_version = self.version_of('xz.tar.gz')
+        compression_source = bb.path(self.source_of('compression.tar.gz'))
 
-        test_dir = local.path(self.testdir)
-        testfiles = [test_dir / x for x in self.testfiles]
-        cp(testfiles, self.builddir)
+        tar('xf', xz_source)
+        tar('xf', compression_source)
 
-        unpack_dir = local.path('xz-{0}'.format(self.version))
-        clang = compiler.cc(self)
-        with local.cwd(unpack_dir):
+        unpack_dir = bb.path(f'xz-{xz_version}')
+        clang = bb.compiler.cc(self)
+        with bb.cwd(unpack_dir):
             configure = local["./configure"]
-            _configure = run.watch(configure)
-            with local.env(CC=str(clang)):
+            _configure = bb.watch(configure)
+            with bb.env(CC=str(clang)):
                 _configure("--enable-threads=no", "--with-gnu-ld=yes",
                            "--disable-shared", "--disable-dependency-tracking",
                            "--disable-xzdec", "--disable-lzmadec",
                            "--disable-lzmainfo", "--disable-lzma-links",
                            "--disable-scripts", "--disable-doc")
 
-            _make = run.watch(make)
+            _make = bb.watch(make)
             _make("CC=" + str(clang), "clean", "all")
 
     def run_tests(self):
-        unpack_dir = local.path('xz-{0}'.format(self.version))
-        xz = wrapping.wrap(unpack_dir / "src" / "xz" / "xz", self)
-        _xz = run.watch(xz)
+        xz_version = self.version_of('xz.tar.gz')
+        unpack_dir = bb.path(f'xz-{xz_version}')
+        xz = bb.wrap(unpack_dir / "src" / "xz" / "xz", self)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable name "xz" doesn't conform to '[a-z_][a-z0-9_]{2,30}$' pattern](https://app.codacy.com/manual/polyjit/benchbuild_2/pullRequest?prid=5694346)